src/util-lua-flowvarlib.c

```diff
@@ -0,0 +1,144 @@
+/* Copyright (C) 2025 Open Information Security Foundation
+ *
+ * You can copy, redistribute or modify this Program under the terms of
+ * the GNU General Public License version 2 as published by the Free
+ * Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * version 2 along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
+ */
+
+#include "suricata-common.h"
+#include "app-layer-protos.h"
+#include "flow-var.h"
+#include "lauxlib.h"
+#include "util-debug.h"
+#include "util-lua-common.h"
+#include "util-lua-flowvarlib.h"
+#include "util-lua.h"
+#include "util-var-name.h"
+#include "detect-lua.h"
+#include "detect-lua-extensions.h"
+
+static const char suricata_flowvar_mt[] = "suricata:flowvar:mt";
+
+static DetectLuaData *GetLuaData(lua_State *luastate)
+{
+    DetectLuaData *ld;
+    lua_pushlightuserdata(luastate, (void *)&luaext_key_ld);
+    lua_gettable(luastate, LUA_REGISTRYINDEX);
+    ld = lua_touserdata(luastate, -1);
+    return ld;
+}
+
+/**
+ * \brief Register a flowvar.
+ *
+ * Ensures that a flowvar exists for the provided name, will be
+ * created if needed.
+ *
+ * The flowvar ID is returned, however as this is most likely to be
+ * used in the scripts "init" method, this ID is unlikely to be used.
+ */
+static int LuaFlowvarRegister(lua_State *L)
+{
+    DetectLuaData *ld = GetLuaData(L);
+    const char *name = luaL_checkstring(L, 1);
+    uint32_t *flowvar_id = lua_newuserdata(L, sizeof(*flowvar_id));
+    *flowvar_id = VarNameStoreRegister(name, VAR_TYPE_FLOW_VAR);
+    if (*flowvar_id == 0) {
+        return luaL_error(L, "failed to register flowvar");
+    }
+    ld->flowvar[ld->flowvars++] = *flowvar_id;
+
+    luaL_getmetatable(L, suricata_flowvar_mt);
+    lua_setmetatable(L, -2);
+
+    return 1;
+}
+
+static int LuaFlowvarGet(lua_State *L)
+{
+    const char *name = luaL_checkstring(L, 1);
+    uint32_t *flowvar_id = lua_newuserdata(L, sizeof(*flowvar_id));
+    *flowvar_id = VarNameStoreLookupByName(name, VAR_TYPE_FLOW_VAR);
+    if (*flowvar_id == 0) {
+        return luaL_error(L, "flowvar does not exist");
+    }
+
+    luaL_getmetatable(L, suricata_flowvar_mt);
+    lua_setmetatable(L, -2);
+
+    return 1;
+}
+
+/**
+ * \brief Return the current value of the flowvar.
+ */
```

Comment:
probably a bit nitty, but we're not actually returning the value, right? We're setting it in the lua state? Why not use that wording in the function docs?

Reply:
Yeah, this is from the perspective of Lua, not C. Should clear that up.